var/spack/repos/builtin/packages/py-pyseer/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyPyseer(PythonPackage):
+    """Python implementation of Sequence Element Enrichment Analysis (SEER)"""
+
+    homepage = "https://github.com/mgalardini/pyseer"
+
+    url = "https://github.com/mgalardini/pyseer/archive/refs/tags/1.3.11.tar.gz"
+
+    version("1.3.11", sha256="06ea2987509f9c1952bbb90e4b59c6f5a4f2ca9e88e7dac5f5cb7f43aa693a1b")
+
+    # build options
+    variant("matplotlib", default=True, description="Enables automatic scree plots")
+    variant("dendropy", default=True, description="Enables calculating phylogeny distances")
+    variant("kmer", default=True, description="Enables k-mer mapping and annotation")
+
+    # build deps
+    depends_on("py-setuptools", type="build")
+
+    # additional deps
+    # https://pyseer.readthedocs.io/en/master/installation.html
+    # also from
+    # github.com/bioconda/bioconda-recipes/blob/master/recipes/pyseer/meta.yaml
+    # the deps are messy and this project is not updated
+    depends_on("py-numpy@1.15.2:", type=("build", "run"))
+    depends_on("py-scipy@1.1.0:", type=("build", "run"))
+    depends_on("py-pandas@0.23.4:", type=("build", "run"))
+    depends_on("py-scikit-learn@0.20.0:", type=("build", "run"))
+    depends_on("py-statsmodels@0.9.0:", type=("build", "run"))
+    depends_on("py-pysam@0.15.3:", type=("build", "run"))
+    depends_on("py-glmnet-python@1.0", type=("build", "run"))
+    depends_on("py-dendropy@4.4.0:", type=("build", "run"))
+    depends_on("py-tqdm@4.20.0:", type=("build", "run"))
+
+    # optional deps
+    depends_on("py-matplotlib@2.1.0:", when="+matplotlib", type=("build", "run"))
+    depends_on("py-dendropy@4.3.0:", when="+dendropy", type=("build", "run"))
+    depends_on("py-pybedtools@0.7.10:", when="+kmer", type=("build", "run"))
+    depends_on("bedops@2.4.9:", when="+kmer", type=("build", "run"))
+    depends_on("bwa", when="+kmer", type=("build", "run"))
```

Comment:
I am a little confused on which basis which dependency was added: 

According to [installation.html](https://pyseer.readthedocs.io/en/master/installation.html) `bedtools` is also a dependency. I guess it was not explicitly defined because it is an implicit dependency of `py-pybedtools`. But is `bedtools2` (the one from `py-pybedtools`) the same as `bedtools` from here`?

On which basis did you choose additional dependencies from [bioconda](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pyseer/meta.yaml). E.g. you have added `bwa` but not `mash` or `python-dateutil`.

Reply:
I didn't include `bedtools` because it's an implicit dependency of `py-pybedtools` and it seemed redundant to list it twice. I can add it back it that's better.

`bwa` is used in a a util (`pyseer/kmer_mapping/bwa.py`). 

`mash` is an unnecessary dependency. It would be useful to have in the same environment as pyseer can interact with its output, but the executable is never directly called by `pyseer`.[ In the `pyseer` tutorial, they just run mash directly from the command line and then use a `pyseer` function on the output.](https://github.com/mgalardini/pyseer/blob/master/docs/tutorial.rst). Again, I can add it if you feel that is better.

I have no idea why `python-dateutil` is listed as a dependency. It's not used in the `pyseer` source code at all beyond the requirements list. I have confirmed that it runs fine without it.

Reply:
Oh, and bedtools and bedtools2 are the same thing. Since v2, bedtools uses a new repo called "bedtools2" but the name in bioconda is still "bedtools".

https://github.com/arq5x/bedtools2

https://github.com/arq5x/bedtools

https://github.com/bioconda/bioconda-recipes/blob/master/recipes/bedtools/meta.yaml

Reply:
Regarding bedtools. I don't know how the policy is for implicit dependencies in spack. I was just asking because it is explicitly listed in the installation description. Maybe @adamjstewart can help here.

For the other dependencies, I think the lesser the better as long as the software still works. Maybe adding comments would help.

Ah, I also just saw that this is also described in the [README.md](https://github.com/mgalardini/pyseer/blob/master/README.md#prerequisites).
So in total all you took from bioconda was the `pysam` version, right? So maybe I would refer to `installation.html` and `README.md` for the dependencies in your comment above and just make a small comment to `pysam` about the version. But this is a little bit of nitpicking here :-)